winter/routing/routing.py

```diff
@@ -40,16 +39,35 @@ def route_put(url_path='', produces: Optional[Tuple[MediaType]] = None, consumes
     return route(url_path, 'PUT', produces=produces, consumes=consumes)
 
 
-def register_route(
-        func,
-        url_path: str,
-        http_method: Optional[str],
-        produces: Optional[Tuple[MediaType]],
-        consumes: Optional[Tuple[MediaType]],
-):
-    assert func not in _route_annotations, f'{func} is already mapped to a route'
-    _route_annotations[func] = RouteAnnotation(url_path, http_method, produces, consumes)
-
-
-def get_route_annotation(func) -> Optional[RouteAnnotation]:
-    return _route_annotations.get(func)
+def get_route_annotation(component_method_or_component) -> Optional[RouteAnnotation]:
+    if isinstance(component_method_or_component, ComponentMethod):
+        annotations = component_method_or_component.annotations
+        return annotations.get_one(RouteAnnotation)
+    elif isinstance(component_method_or_component, Component):
+        annotations = component_method_or_component.annotations
+        route_annotations = annotations.get(RouteAnnotation)
+        return route_annotations[0] if route_annotations else None
+    else:
+        raise AssertionError(f'invalid {component_method_or_component}')
+
+
+def get_route(method: ComponentMethod) -> Route:
+    route = method.annotations.get_one_or_none(Route)
```

Comment:
Why do we need such kind of cache? It confuses, because I'm expecting we won't call `get_route` a lot of times and only at the stage on configuration.